src/librustdoc/html/markdown.rs

```diff
@@ -1413,6 +1416,52 @@ impl MarkdownItemInfo<'_> {
     }
 }
 
+pub(crate) fn markdown_split_summary_and_content(
```

Comment:
This should have a comment explaining what it does. Also, it should probably be a method on the `Markdown` struct? It's basically a different version of `to_string`.

```rust
/// Convert markdown to (summary, remaining) HTML.
///
/// - The summary is the first top-level Markdown element (usually a paragraph, but potentially any block).
/// - The remaining docs contain everything after the summary.
```

Reply:
Good idea for the docs. I don't have an opinion on whether or not it should be a method so I'll just make a method since you seem to favor it. :+1: